GitCommands/Git/GitModule.cs

```diff
@@ -1526,7 +1526,23 @@ public string CheckoutFiles(IEnumerable<string> fileList, string revision, bool
             if (files.IsNullOrWhiteSpace())
                 return string.Empty;
 
-            return RunGitCmd("checkout " + force.AsForce() + revision.Quote() + " -- " + files);
+            //Special handling for artificial commits
+            if (revision == GitRevision.IndexGuid)
+            {
+                revision = "";
+            }
+            else if (revision == GitRevision.UnstagedGuid)
+            {
+                Debug.Assert(false, "Unexpectedly reset to unstaged - should be blocked in GUI");
+                //However, not an error to user, just nothing happens
+                return "";
+            }
```

Comment:
I'd move this condition up, so the code reads more naturally:

```
            if (files.IsNullOrWhiteSpace())
                return string.Empty;

            //Special handling for artificial commits
            if (revision == GitRevision.UnstagedGuid)
            {
                Debug.Assert(false, "Unexpectedly reset to unstaged - should be blocked in GUI");
                //However, not an error to user, just nothing happens
                return "";
            }

            if (revision == GitRevision.IndexGuid)
            {
                revision = "";
            }
            else
            {
                revision = revision.QuoteNE();
            }
```

Reply:
Ok